File: backend/app/api/health.py
"""Health-check endpoint (A1 foundation).

``GET /health`` → ``{"status": "ok", ...}``. The contracted field is ``status``; ``service``
and ``phase`` are additive metadata (API_CONTRACTS §9). Mounted at the ROOT path (not under
``/api``) by convention for liveness probes and cloud load balancers — see
docs/ARCHITECTURE.md §11 (cloud-ready / 12-factor). Never exposes secrets.
"""

from __future__ import annotations

from fastapi import APIRouter

from app.schemas.common import HealthResponse

router = APIRouter(tags=["health"])


@router.get("/health", response_model=HealthResponse, summary="Liveness probe")
def health() -> HealthResponse:
    """Return a simple liveness response confirming the backend is running."""
    return HealthResponse(
        status="ok", service="routewise-agentic-backend", phase="A1-foundation"
    )

File: frontend/src/services/api/health.ts
/**
 * `GET /health` — liveness probe (A1 brief §12). Mounted at the ROOT path, not under `/api`
 * (backend/app/api/health.py; ARCHITECTURE §11 cloud-ready). Proves the backend is reachable.
 */

import type { HealthResponse } from '../../types/api';
import { request } from './client';

/** Fetch backend health. @throws {ApiError} when unreachable or non-2xx. */
export function getHealth(): Promise<HealthResponse> {
  return request<HealthResponse>('/health', { method: 'GET' });
}
